server.ts

```diff
@@ -26,11 +26,46 @@ app.use(cors()) //add CORS support to each following route handler
 const client = new Client(dbConfig);
 client.connect();
 
-app.get("/", async (req, res) => {
-  const dbres = await client.query('select * from categories');
-  res.json(dbres.rows);
+app.get("/dogs", async (req, res) => {
+  const dbres = await client.query('select breed from dogs');
+  res.set('Access-Control-Allow-Origin', '*')
+  res.status(200).json(dbres.rows);
 });
 
+app.get("/topten", async (req, res) => {
+  const dbres = await client.query(`SELECT * FROM dogs ORDER BY votes DESC, breed ASC LIMIT 10`)
+  res.set('Access-Control-Allow-Origin', '*')
+  res.status(200).json(dbres.rows)
+})
+
+app.post("/", async (req, res) => {
+  try{
+  if(req.body.breedNames.length>0){
+  await client.query('insert into dogs (votes, breed) values ($1,$2) on conflict (breed) do nothing', ['0', req.body.breedNames[0]])
+  await client.query('insert into dogs (votes, breed) values ($1,$2) on conflict (breed) do nothing', ['0', req.body.breedNames[1]])
+  res.set('Access-Control-Allow-Origin', '*')
+  res.status(200).send({success: true, data: req.body.breedNames})
+  }
+  else{
+    res.status(100).send("connection with server secured")
+  }}
+  catch(err){
+    res.send(err.message)
+  }
+});
+
+app.put("/" ,async (req, res) => {
+  try {
+    const response = await client.query('update dogs set votes = votes + 1 where breed = $1 RETURNING * ', [req.body.breedName])
```

Comment:
Perhaps a more descriptive variable name could be used here: `[req.body.breedName]`